File: PerformanceTesting/TextEditCell.cs
using System;
using MonoTouch.UIKit;
using System.Drawing;
using MonoTouch.Foundation;


namespace PerformanceTesting
{
	public class TextEditCell : UITableViewCell
	{
		UITextField _textField;
		
		public TextEditCell (string resuseIdentifier) : base (UITableViewCellStyle.Value1, "resuseIdentifier")
		{
			this.SelectionStyle = UITableViewCellSelectionStyle.None;
			
			_textField = new UITextField (new RectangleF (150, 10, 160, 30));
	        _textField.AdjustsFontSizeToFitWidth = true;
	        _textField.TextColor = UIColor.Black;
			_textField.Placeholder = "enter name";
			_textField.KeyboardType = UIKeyboardType.Default;
            _textField.ReturnKeyType = UIReturnKeyType.Done;
			
			_textField.BackgroundColor = UIColor.Clear;
	        _textField.AutocorrectionType = UITextAutocorrectionType.No; // no auto correction support
	        _textField.AutocapitalizationType = UITextAutocapitalizationType.None; // no auto capitalization support
	        _textField.TextAlignment = UITextAlignment.Right;
	        _textField.Tag = 0;
	        _textField.WeakDelegate = this;
			_textField.AutoresizingMask = UIViewAutoresizing.All;
	
	        //_textField.ClearButtonMode = UITextFieldViewMode.WhileEditing; // no clear 'x' button to the right
			_textField.ClearButtonMode = UITextFieldViewMode.Never; // no clear 'x' button to the right
			_textField.Enabled= true;
			_textField.Font = this.DetailTextLabel.Font;
			
			this.AddSubview (_textField);
		}
		
		public override void LayoutSubviews ()
		{
			base.LayoutSubviews ();	
			
			float x = this.Subviews[0].Frame.X + this.TextLabel.Frame.X;
			
			RectangleF rect = this.TextLabel.Bounds;
			
			rect.X = x + rect.Width + 10;
			rect.Width = this.Bounds.Width - (rect.X + x);
			rect.Y = 10; rect.Height = 30;
			
			//rect.X += 150; rect.Width -= 150;
			_textField.Frame = rect;
			_textField.Font = this.DetailTextLabel.Font;
			_textField.TextColor = this.DetailTextLabel.TextColor;
			//_textField.BackgroundColor = UIColor.Black;
			//this.TextLabel.BackgroundColor = UIColor.Black;
		}	

		public UITextField TextField 
		{
			get 
			{
				return _textField;
			}
		}
		
		[Export ("textFieldDidEndEditing:")]
		public virtual void EditingEnded (UITextField textField)
		{
			DeviceInfo.OwnerName = textField.Text;
		}
		
		[Export ("textFieldShouldReturn:")]
		public virtual bool ShouldReturn (UITextField textField)
		{
			textField.ResignFirstResponder ();
			return false;
		}
	}
}

File: CPerformanceTesting/CPerformanceTesting/TextEditCell.m
//
//  TextEditCellCell.m
//  CPerformanceTesting
//
//  Created by William Holmes on 6/23/12.
//  Copyright (c) 2012 Slap Holmes Productions. All rights reserved.
//

#import "TextEditCell.h"
#import "DeviceInfo.h"

@implementation TextEditCell

@synthesize textField;

- (id)initWithReuseIdentifier:(NSString *)identifier
{
    self = [super initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:identifier];
    if (self) {
        self.selectionStyle = UITableViewCellSelectionStyleNone;
        
        CGRect rect = {150, 10, 160, 30};
        textField = [[UITextField alloc]initWithFrame:rect];
        textField.adjustsFontSizeToFitWidth = YES;
        textField.textColor = [UIColor blackColor];
        textField.placeholder = @"enter name";
        textField.keyboardType = UIKeyboardTypeDefault;
        textField.returnKeyType = UIReturnKeyDone;
        
        textField.backgroundColor = [UIColor clearColor];
        textField.autocorrectionType = UITextAutocorrectionTypeNo; // no auto correction support
        textField.autocapitalizationType = UITextAutocapitalizationTypeNone; // no auto capitalization support
        textField.textAlignment = UITextAlignmentRight;
        textField.tag = 0;
        textField.delegate = self;
        textField.autoresizingMask = UIViewAutoresizingFlexibleWidth;
        
        textField.clearButtonMode = UITextFieldViewModeNever; // no clear 'x' button to the right
        textField.enabled= YES;
        textField.font = self.detailTextLabel.font;
        
        [self addSubview: textField];
    }
    return self;
}

-(void) layoutSubviews
{
    [super layoutSubviews];
    UIView* sView = [self.subviews objectAtIndex:0];
    
    float x = sView.frame.origin.x + self.textLabel.frame.origin.x;
    
    CGRect rect = self.textLabel.bounds;
    
    rect.origin.x = x + rect.size.width + 10;
    rect.size.width = self.bounds.size.width - (rect.origin.x + x);
    rect.origin.y = 10; rect.size.height = 30;
    
    textField.frame = rect;
    textField.font = self.detailTextLabel.font;
    textField.textColor = self.detailTextLabel.textColor;
}

- (void)textFieldDidEndEditing:(UITextField *)ltextField
{
    [DeviceInfo current].ownerName = textField.text;
}

- (BOOL)textFieldShouldReturn:(UITextField *)ltextField
{
    [textField resignFirstResponder];
    return NO;
}

@end
